File: src/main/java/instance/os/InstanceCost.java
package instance.os;

import instance.Node;
import se.sics.kompics.address.Address;
import se.sics.kompics.network.Message;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-04-28
 *
 */

public class InstanceCost extends Message {

	private static final long serialVersionUID = 3300052765351863627L;
	private String totalCost;
	private Node node;
	private String periodicCost;

	public InstanceCost(Address source, Address destination, Node node, String totalCost, String periodicCost) {
		super(source, destination);
		this.node = node;
		this.totalCost = totalCost;
		this.periodicCost = periodicCost; 
	}

	public String getTotalCost() {
		return totalCost;
	}

	public String getPeriodicCost() {
		return periodicCost;
	}

	public Node getNode() {
		return node;
	}

}

File: src/main/java/instance/os/InstanceCost.scala
package instance.os

import se.sics.kompics.network.Message
import se.sics.kompics.address.Address
import instance.Node
import reflect.BeanProperty

/**
 * @author Amir Moulavi
 * @date 2011-07-16
 *
 */
class InstanceCost(source:Address,
                   destination:Address,
                   @BeanProperty var node:Node,
                   @BeanProperty var totalCost:String,
                   @BeanProperty var periodicCost:String)
  extends Message(source, destination)